Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/VaccinationRepository.kt

```diff
@@ -434,9 +434,50 @@ class VaccinationRepository @Inject constructor(
 
     suspend fun replaceCertificate(
         certificateToReplace: VaccinationCertificateContainerId,
-        newCertificateQrCodeQrCode: VaccinationCertificateQRCode
+        newCertificateQrCode: VaccinationCertificateQRCode
     ) {
         // TO_DO("https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11940")
+        // TODO Update after migration
+        internalData.updateBlocking {
+            // Recycle
+            val matchingPerson = singleOrNull { it.findVaccination(certificateToReplace) != null }
+
+            if (matchingPerson == null) {
+                Timber.tag(TAG).w("recycleCertificate couldn't find %s", certificateToReplace)
+                return@updateBlocking this
+            }
+
+            val toUpdateVaccination = matchingPerson.findVaccination(certificateToReplace)!!
+            val newVaccination = toUpdateVaccination.copy(recycledAt = timeStamper.nowUTC)
+            newVaccination.qrCodeExtractor = qrCodeExtractor
+
+            if (matchingPerson.data.vaccinations.any { it.qrCodeHash == newCertificateQrCode.hash }) {
+                Timber.tag(TAG).e("Certificate is already registered: %s", newCertificateQrCode.hash)
+                throw InvalidVaccinationCertificateException(ALREADY_REGISTERED)
+            }
+
+            val newCertificate = newCertificateQrCode.toVaccinationContainer(
+                scannedAt = timeStamper.nowUTC,
+                qrCodeExtractor = qrCodeExtractor,
+                certificateSeenByUser = false,
+            )
+
+            val newPersonData = matchingPerson.data.copy(
+                vaccinations = matchingPerson.data.vaccinations.minus(toUpdateVaccination)
+                    .plus(newVaccination)
+                    .plus(newCertificate)
+            )
+
+            val modifiedPerson = matchingPerson.copy(
+                data = newPersonData,
+                certificateStates = newPersonData.getStates()
+            )
+
+            this.toMutableSet().apply {
```

Comment:
NIT
this.minus(matchingPerson).plusElement(modifiedPerson)

Reply:
this is temporary, see the comment above , it will be changed anyway